### PR TITLE
Moving ml docker from Pip to Poetry and updating onnx to 1.17.0 version

### PR DESCRIPTION
**Pull request by rshunim**
## Status
Ready

## Related Issues
Related: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-12067)

## Description
Moving the ml docker to poetry and updating onnx to 1.17.0

### CODE REVIEW

**Comment by rshunim** [comment]
Should wait for @jlevypaloalto's review

File: docker/ml/Dockerfile [hunk 1]
```diff
@@ -12,25 +12,21 @@ RUN unzip /distilled_bert_base_uncased_tokenizer.zip -d /ml \
...
-&& apt-get purge -y --auto-remove gcc \
-  python3-dev gfortran wget git g++ pkg-config libhdf5-dev pipenv \
-&& apt-get upgrade -y \
-&& apt install -y perl libsqlite3-0
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends gcc python3-dev pipenv gfortran wget git g++ pkg-config libhdf5-dev
+RUN pip install --no-cache-dir -r requirements.txt
+RUN apt-get purge -y --auto-remove gcc python3-dev gfortran wget git g++ pkg-config libhdf5-dev
+RUN apt-get upgrade -y
+RUN apt install -y perl libsqlite3-0
```
Comment (dorschw) [inline docker/ml/Dockerfile:23]: now when we're done debugging, please put these back as one `RUN`

File: docker/ml/Dockerfile [hunk 2]
```diff
@@ -12,25 +12,21 @@ RUN unzip /distilled_bert_base_uncased_tokenizer.zip -d /ml \
...
 
 COPY demisto_ml-0.2.tar.gz .
 RUN pip install demisto_ml-0.2.tar.gz
 RUN rm -rf /var/lib/apt/lists/*
 RUN python -m spacy download en_core_web_sm
-RUN mkdir /ml
+RUN mkdir -p /ml/nltk_data && chmod -R 777 /ml/nltk_data
```
Comment (dorschw) [inline docker/ml/Dockerfile:29]: why `777`?

File: docker/ml/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+[tool.poetry]
+name = "ml"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (dorschw) [inline docker/ml/pyproject.toml:5]:
```suggestion
authors = ["Demisto <demisto@demisto.com>"]
```

File: docker/ml/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+[[tool.poetry.source]]
+name = "pytorch"
+url = "https://download.pytorch.org/whl/cpu/"
+priority = "explicit"
+
+
+[tool.poetry.dependencies]
+python = "~3.10"
+fasttext = "*"
+numpy = "==1.26.4" # TODO - add docs why it cannot be updated
```
Comment (dorschw) [inline docker/ml/pyproject.toml:18]: todo

**Review by dorschw — COMMENTED** [review]
nice

**Review by dorschw — APPROVED** [review]

File: docker/ml/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+[[tool.poetry.source]]
+name = "pytorch"
+url = "https://download.pytorch.org/whl/cpu/"
+priority = "explicit"
+
+
+[tool.poetry.dependencies]
+python = "~3.10"
+fasttext = "*"
+numpy = "==1.26.4" # cannot be updated more than this version due to other dependencies
```
Comment (jlevypaloalto) [inline docker/ml/pyproject.toml:18]:
```suggestion
numpy = "==1.26.4"  # Upgrading to version 2.0.0 may introduce known performance issues, it must be approached with caution.
```

**Review by jlevypaloalto — COMMENTED** [review]
The switch to poetry is fantastic.
Let's link the content PR where we tested all the ML scripts that use this image.

**Comment by rshunim** [comment]
The Content PR https://github.com/demisto/content/pull/37612
The pre-commit passed with running the all UTs with `devdemisto/ml:1.0.0.118104`

**Comment by rshunim** [comment]
last link to the Content PR
https://github.com/demisto/content/pull/37612

**Comment by rshunim** [comment]
https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/1868975

**Pull request merged**